plugin/fallback/upstream_utils.go

```diff
@@ -0,0 +1,50 @@
+// Package fallback implements a fallback plugin for CoreDNS
+package fallback
+
+import (
+	"github.com/coredns/coredns/plugin"
+	"github.com/coredns/coredns/plugin/proxy"
+)
+
+// upstreamMapper manages map of rcode to upstream
+type upstreamMapper interface {
```

Comment:
I don't understand why we need this separate interface and struct. It doesn't really do anything more than the map did already. Just seems like extra stuff.

Reply:
Yeah.... I was going to use that to test setup().

OK, I'll revert it back to the original rules map.